src/screens/Home/components/Technologies/TechnologyCard.tsx

```diff
@@ -0,0 +1,230 @@
+
+interface DataCardProps {
+  id: number
+  title: string
+  status: 'public' | 'private'
+  date: string
+  image: string
+  category: {
+    name: string
+  }
+}
+interface TechnologyCardProps {
+  data?: DataCardProps | {}
+  style?: StyleProp<any>
+  loading?: boolean
+  navigation: StackNavigationProp<any, any>
+};
+
+interface FavoriteProps {
+  favorite?: boolean
+};
+
+const CardWrapper = styled(View)`
+  width: 248px;
+  height: 381px;
+`;
+
+const CardContainer = styled(View)`
+  padding: 16px;
+`;
+
+const CardImage = styled(View)`
+  backgroundColor: #f5f5f5;
+  width: 216px;
+  height: 216px;
+  borderRadius: 8px;
+`;
+
+const Title = styled(DefaultText)`
+  height: 55px;
+  fontFamily: Rubik_500Medium;
+  fontWeight: 500;
+  fontSize: 18px;
+  lineHeight: 27px;
+  textAlign: center;
+  marginTop: 16px;
+`;
+
+const StatusWrapper = styled(View)`
+  marginTop: 8px;
+  marginBottom: 16px;
+  flexDirection: row;
+  alignItems: center;
+  justifyContent: center;
+`;
+
+const Status = styled(DefaultText)`
+  fontFamily: Rubik_500Medium;
+  fontWeight: 500;
+  fontSize: 12px;
+  lineHeight: 18px;
+  textAlign: center;
+  color: ${Colors.primary};
+`;
+
+const StatusIcon = styled(Feather)`
+  marginRight: 6px;
+`;
+
+const DateWrapper = styled(View)`
+  flexDirection: row;
+  alignItems: center;
+  justifyContent: center;
+`;
+
+const Date = styled(DefaultText)`
+  fontFamily: Rubik_500Medium;
+  fontWeight: 500;
+  fontSize: 12px;
+  lineHeight: 18px;
+  textAlign: center;
+  color: #a5a5a5;
+`;
+
+const DateIcon = styled(Feather)`
+  marginRight: 6px;
+`;
+
+const Actions = styled(View)`
+  width: 100%;
+  height: 26px;
+  flexDirection: row;
+  alignItems: center;
+  justifyContent: space-between;
+  position: absolute;
+  bottom: 16px;
+  zIndex: 1;
+`;
+
+const LabelWrapper = styled(View)`
+  backgroundColor: #ccede7;
+  width: auto;
+  height: 26px;
+  paddingVertical: 4px;
+  paddingHorizontal: 8px;
+  borderTopRightRadius: 8px;
+  borderBottomRightRadius: 8px;
+`;
+
+const LabelText = styled(DefaultText)`
+  fontFamily: Rubik_500Medium;
+  fontWeight: 500;
+  fontSize: 12px;
+  lineHeight: 18px;
+  color: ${Colors.primary};
+`;
+
+const Label = (): JSX.Element => (
+  <LabelWrapper>
+    <LabelText>Semiárido</LabelText>
+  </LabelWrapper>
+);
+
+const FavoriteButton = styled(TouchableOpacity)`
+  marginRight: 18px;
+`;
+
+const FavoriteIcon = styled(FontAwesome5)``;
+
+const Favorite = ({ favorite = false }: FavoriteProps): JSX.Element => {
+  const [state, setState] = useState(favorite);
+  const animatePulse = new Animated.Value(1);
+  const scale = animatePulse.interpolate({
+    inputRange: [0, 1],
+    outputRange: [1.1, 1]
+  });
+
+  useEffect(() => {
+    Animated.timing(animatePulse, {
+      toValue: state ? 0 : 1,
+      duration: state ? 100 : 50,
+      easing: Easing.linear,
+      useNativeDriver: true
+    }).start();
+  }, [state]);
+
+  return (
+    <FavoriteButton onPress={() => setState(!state)}>
```

Comment:
```suggestion
    <FavoriteButton onPress={() => setState(previousState => !previousState)}>
```
Nesse caso, como o setState é assíncrono, pode acontecer um probleminha qnd o usuário clicar rapidamente: ele não fazer as atualizações em sequencia e acabar em um estado inconsistente. Ex.: tá não favoritado, o usuário clica 2 vezes. Ao invés de voltar para não favoritado, ficar como favoritado. Pra garantir que isso não aconteçã, temos de usar uma função (ela garante o retorno sempre do último estado alterado).